### PR TITLE
Re-order function arguments

### PR DESCRIPTION
**Pull request by vmcj**
PSR wants the argument on a separate line.

This is the last fix needed for now, to remove most of the annotations unrelated to the changes in PRs.

This should be merged after #1846

### CODE REVIEW

**Review by nickygerritsen — APPROVED** [review]

File: judge/judgedaemon.main.php [hunk 1]
```diff
@@ -275,16 +275,23 @@ function read_judgehostlog(int $n = 20) : string
...
 function fetch_executable(
-    string $workdirpath, string $type, string $execid, string $hash, int $judgeTaskId, bool $combined_run_compare = false) : array
-{
+    string $workdirpath,
+    string $type,
+    string $execid,
+    string $hash,
+    int $judgeTaskId,
+    bool $combined_run_compare = false
+) : array {
```
Comment (eldering) [inline judge/judgedaemon.main.php:284]: The brace should go on a new line.

File: judge/judgedaemon.main.php [hunk 2]
```diff
@@ -304,8 +311,12 @@ function fetch_executable(
...
 // - optional extra build log.
 function fetch_executable_internal(
-    string $workdirpath, string $type, string $execid, string $hash, bool $combined_run_compare = false) : array
-{
+    string $workdirpath,
+    string $type,
+    string $execid,
+    string $hash,
+    bool $combined_run_compare = false
+) : array {
```
Comment (eldering) [inline judge/judgedaemon.main.php:319]: The brace should go on a new line.

File: judge/judgedaemon.main.php [hunk 3]
```diff
@@ -947,9 +972,14 @@ function registerJudgehost(string $myhost): void
...
-                 ?int $judgeTaskId = null, ?string $extra_log = null): void
-{
+function disable(
+    string $kind,
+    string $idcolumn,
+    $id,
+    string $description,
+    ?int $judgeTaskId = null,
+    ?string $extra_log = null
+): void {
```
Comment (eldering) [inline judge/judgedaemon.main.php:982]: The brace should go on a new line.

File: judge/judgedaemon.main.php [hunk 4]
```diff
@@ -975,12 +1005,14 @@ function disable(string $kind, string $idcolumn, $id, string $description,
 
 function read_metadata(string $filename): ?array
 {
-    if (!is_readable($filename)) return null;
+    if (!is_readable($filename)) {
+        return null;
+    }
```
Comment (eldering) [inline judge/judgedaemon.main.php:1010]: IMO this doesn't improve readability as now 2 more lines are spent on the same code, so 2 fewer lines fit on your screen.

File: judge/judgedaemon.main.php [hunk 5]
```diff
@@ -1015,9 +1047,14 @@ function cleanup_judging(string $workdir) : void
...
-    int $output_storage_limit): bool
-{
+function compile(
+    array $judgeTask,
+    string $workdir,
+    string $workdirpath,
+    array $compile_config,
+    string $cpuset_opt,
+    int $output_storage_limit
+): bool {
```
Comment (eldering) [inline judge/judgedaemon.main.php:1057]: The brace should go on a new line.

File: webapp/src/Command/ScoreboardMergeCommand.php [hunk 6]
```diff
@@ -307,7 +308,14 @@ protected function execute(InputInterface $input, OutputInterface $output): int
...
         $data = $this->scoreboardService->getScoreboardTwigData(
-            null, null, '', false, true, true, $contest, $scoreboard
+            null,
+            null,
+            '',
+            false,
+            true,
+            true,
+            $contest,
+            $scoreboard
```
Comment (eldering) [inline webapp/src/Command/ScoreboardMergeCommand.php:318]: Why not simply put everything on a single line? This wastes a lot of lines with little content.

File: webapp/src/Controller/API/JudgehostController.php [hunk 7]
```diff
@@ -918,9 +957,15 @@ protected function giveBackJudging(int $judgingId, ?Judgehost $judgehost): void
...
             }
 
-            $this->dj->auditlog('judging', $judgingId, 'given back'
-                . ($judgehost === null ? '' : ' for judgehost ' . $judgehost->getHostname()), null,
-                $judgehost === null ? null : $judgehost->getHostname(), $judging->getContest()->getCid());
+            $this->dj->auditlog(
+                'judging',
+                $judgingId,
+                'given back'
+                . ($judgehost === null ? '' : ' for judgehost ' . $judgehost->getHostname()),
```
Comment (eldering) [inline webapp/src/Controller/API/JudgehostController.php:964]: This is part of the same argument.

File: webapp/src/Controller/API/SubmissionController.php [hunk 8]
```diff
@@ -432,8 +443,19 @@ public function addSubmissionAction(Request $request, ?string $id): Response
...
+            $problem->getContest(),
+            $language,
+            $files,
+            'API',
+            null,
+            null,
+            $entryPoint,
+            $submissionId,
+            $time,
+            $message
```
Comment (eldering) [inline webapp/src/Controller/API/SubmissionController.php:458]: A lot of lines wasted here.

File: webapp/src/Service/EventLogService.php [hunk 9]
```diff
@@ -686,35 +693,47 @@ public function initStaticEvents(Contest $contest): void
...
 
                 // Special case 'contests' since it is a single object:
                 if ($endpoint === 'contests') {
                     if (!is_array($data) || is_numeric(array_keys($data)[0])) {
-                        throw new Exception(sprintf("EventLogService::initializeStaticEvents Endpoint '%s' did not return a JSON object.",
-                                                    $endpoint));
+                        throw new Exception(sprintf(
+                            "EventLogService::initializeStaticEvents Endpoint '%s' did not return a JSON object.",
+                            $endpoint
+                        ));
```
Comment (eldering) [inline webapp/src/Service/EventLogService.php:708]: Just simplify this to remove the `sprintf` and put it on a single line.

File: judge/judgedaemon.main.php (hunk 4 above)
Comment (nickygerritsen) [reply judge/judgedaemon.main.php:1010]: I personally really *do* like this (and similar) chnages. I don't like that one line does a check and an action...

File: judge/judgedaemon.main.php (hunk 1 above)
Comment (vmcj) [reply judge/judgedaemon.main.php]: Would that not take an extra unneeded line? I'll take a look why phpcs did not warn on this.

File: webapp/src/Command/ScoreboardMergeCommand.php (hunk 6 above)
Comment (meisterT) [reply webapp/src/Command/ScoreboardMergeCommand.php:318]: I agree that there is no benefit in splitting this over multiple lines. Also when looking at https://www.php-fig.org/psr/psr-2/#46-method-and-function-calls the splitting is a `MAY` not a `MUST`.

File: judge/judgedaemon.main.php (hunk 4 above)
Comment (meisterT) [reply judge/judgedaemon.main.php]: I really prefer always having braces `{}` for `if` statements.

**Comment by vmcj** [comment]
I don't think this gets approval and my other PR removes those checks from the PSR checks.

**Pull request closed without merging**